pkg/images/defaults/flavor.go

```diff
@@ -258,8 +258,27 @@ func GetImageFlavorByName(flavorName string, isReleaseBuild bool) (ImageFlavor,
 	if err := CheckImageFlavorName(flavorName, isReleaseBuild); err != nil {
 		return ImageFlavor{}, err
 	}
-	f := imageFlavorMap[flavorName]
-	return f.constructorFunc(), nil
+
+	return getImageFlavorByName(flavorName), nil
+}
+
+// GetImageFlavorNameFromEnv returns the value of the environment variable (ROX_IMAGE_FLAVOR)
+// providing development_build as default if no RealeseBuild and environment variable not set
+// This function will panic if running a ReleaseBuild and ROX_IMAGE_FLAVOR is not available
```

Comment:
If this panic happen at runtime the user can't fix it.
As we are setting this value in the user-agent at runtime and not build time it could lead to a CrashLoopBackoff for the operator.

Can we get the image flavor safely? I would have expected it to be set on build and not read by the environment var.

@msugakov thoughts on this?

Reply:
Thanks for the explanation @msugakov

Reply:
@johannes94 Nothing to do here, this looks good.